src/components/Microsoft.DotNet.UpgradeAssistant.Analysis/AnalyzeResultWriter.cs

```diff
@@ -49,19 +49,38 @@ await foreach (var ar in analyzeResultDefinitions)
                         {
                             Name = ar.Name,
                             SemanticVersion = ar.Version,
-                            InformationUri = ar.InformationURI,
-                            Rules = analyzeResults.GroupBy(x => x.RuleId).Select(a => new ReportingDescriptor()
-                            {
-                                Id = a.Key,
-                                FullDescription = new() { Text = a.First().RuleName, },
-                            }).ToList(),
+                            InformationUri = ar.InformationUri,
+                            Rules = analyzeResults.GroupBy(x => x.RuleId).Select(ExtractRule).ToList(),
                         },
                     },
                     Results = ExtractResults(analyzeResults),
                 };
             }
         }
 
+        private static ReportingDescriptor ExtractRule(IGrouping<string, AnalyzeResult> analyzeResults)
+        {
+            var analyzeResult = analyzeResults.First();
+
+            var rule = new ReportingDescriptor
+            {
+                Id = analyzeResult.RuleId,
+                HelpUri = analyzeResult.HelpUri
+            };
+
+            if (string.IsNullOrWhiteSpace(analyzeResult.FullDescription))
+            {
+                rule.FullDescription = new() { Text = analyzeResult.RuleName };
```

Comment:
this if is not setting the rule.Name

Reply:
it is by design.
Rules generated by Roslyn does not have a specific RuleName, with that, i'm keeping the old behavior.

Reply:
ah I forgot that we are not updating the roslyn parts here.